File: leetcode/2025/Feb/3174.ClearDigits/solution.py
# -*- encoding: utf-8 -*-
"""
@Introduce  : 
@File       : solution.py
@Author     : ryrl
@Email      : ryrl970311@gmail.com
@Time       : 2025/02/10 18:44
@Description: 
"""

class Solution:
    def clearDigits(self, s: str) -> str:
        
        digits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']

        result = []
        for val in s:
            if val in digits and result:  # val.isdigit() and result
                result.pop()
            else:
                result.append(val)
        return ''.join(result)
                

if __name__ == '__main__':

    s = 'abc'
    s[0].isdigit()

File: leetcode/2025/Feb/3174.ClearDigits/solution.rs
// @Introduce  :
// @File       : solution.rs
// @Author     : ryrl
// @Email      : ryrl970311@gmail.com
// @Time       : 2025/02/10 19:15
// @Description:

impl Solution {
    pub fn clear_digits(s: String) -> String {
        let mut result = Vec::new();

        for val in s.chars() {
            if val.is_ascii_digit() && !val.is_empty() {
                result.pop();
            } else {
                result.push(val);
            }
        }
        result.into_iter().collect()
    }
}
